apps/aering/src/aer_abi.erl

```diff
@@ -12,15 +12,28 @@
 
 -export([create_calldata/3]).
 
+-spec create_calldata(binary(), string(), string()) -> binary() | {error, argument_syntax_error}.
+
 create_calldata(Contract, Function, Argument) ->
     %% TODO: check that function exists in cotract.
     FunctionHandle = encode_function(Contract, Function),
+    case aer_constants:string(Argument) of
+        {ok, ParsedArgument} ->
+            encode_call(FunctionHandle, ParsedArgument);
+        {error, _} ->
+            {error, argument_syntax_error}
+    end.
+
+encode_call(FunctionHandle, ArgumentAst) ->
```

Comment:
You need to agree with compiler on the format. You expect ("add", (1, 2)) but compiler expects ("add", 1, 2).

Reply:
Yes that has to be fixed. I asked a question about this in the ring channel at some point.
My impression is that ring functions always takes one argument: a tuple.

Reply:
It takes a tuple function name and then one item per argument.... but this can be fixed later